File: C#/20. Valid Parentheses.cs
public bool IsValid(string s) {
	Stack<char> stack = new Stack<char>();
	foreach(char c in s){
		if(c == ']' || c== ')' || c=='}'){
			if(stack.Count() == 0) return false;
			if(c == ')' && stack.Peek() != '(') return false;
			if(c == ']' && stack.Peek() != '[') return false;
			if(c == '}' && stack.Peek() != '{') return false;
			stack.Pop();
		}else{
			stack.Push(c);
		}
	}
	if(stack.Count != 0) return false;
	return true;
}

File: Java/20. Valid Parentheses.java
public boolean isValid(String s) {
	Stack<Character> stack = new Stack<Character>();
	for(int i = 0; i < s.length(); i++){
		if(s.charAt(i) == ')' || s.charAt(i) == ']' || s.charAt(i) == '}'){
			if(stack.isEmpty()) return false;
			if(s.charAt(i) == ')' && stack.peek() != '(') return false;
			if(s.charAt(i) == ']' && stack.peek() != '[') return false;
			if(s.charAt(i) == '}' && stack.peek() != '{') return false;
			stack.pop();
		}else{
			stack.push(s.charAt(i));
		}
	}
	if(stack.isEmpty()) return true;
	return false;
}